src/fhirtypes/ElementDefinition.ts

```diff
@@ -1889,6 +1892,60 @@ export class ElementDefinition {
     return [];
   }
 
+  /**
+   * Returns an array of an ElementDefinition's unfolded children.
+   * @param {ElementDefinition} targetElement - The ElementDefinition being unfolded
+   * @returns {ElementDefinition[]} An array of the targetElement's children, with the IDs altered and
+   * the original property re-captured.
+   */
+
+  private cloneChildren(targetElement: ElementDefinition): ElementDefinition[] {
+    return targetElement?.children().map(e => {
+      const eClone = e.clone();
+      eClone.id = eClone.id.replace(targetElement.id, this.id);
+      eClone.structDef = this.structDef;
+      eClone.captureOriginal();
+      return eClone;
+    });
+  }
+
+  /**
+   * Checks a StructureDefinition's differential to determine if the profile element extension has been used on
+   * an element.
+   * @param {Object} profileJson - The json representation of this ElementDefinition's structDef
+   * @returns {boolean} True if the profile element extension is found on this elements profile property, false
+   * if the extension is not found
+   */
+  private hasProfileElementExtension(profileJson: any): boolean {
+    const elementName = this.getContentReferenceId();
+
+    const elementType = profileJson.differential.element.find(
+      (element: any) => element.id === elementName
+    )?.type?.[0];
+
+    // If the type property is not present in the differential, the profile-element extension is not present
+    if (!elementType) {
+      return false;
+    }
+
+    const profileCanonical = elementType.profile?.[0];
+    let extensionUrl, targetElement: string;
+
+    // If there is no _profile array, the profile-element extension is not present
+    if (elementType._profile) {
+      extensionUrl = elementType._profile[0].extension[0].url;
+      targetElement = elementType._profile[0].extension[0].valueString;
+    } else {
+      return false;
+    }
```

Comment:
Is it okay to only be checking `profile[0]` and `_profile[0]` here, or is it possible that the profile element extension is at another index in these arrays?

Reply:
That could certainly create some problems, I'll change this so that the extension will be found even if it's not the first element in the `_profile` array.

Reply:
Addressed in `b4d43c8`